d2common/d2fileformats/d2ds1/ds1.go

```diff
@@ -39,232 +27,252 @@ const (
 
 // DS1 represents the "stamp" data that is used to build up maps.
 type DS1 struct {
-	Files                      []string            // FilePtr table of file string pointers
-	Objects                    []Object            // Objects
-	Tiles                      [][]TileRecord      // The tile data for the DS1
-	SubstitutionGroups         []SubstitutionGroup // Substitution groups for the DS1
-	Version                    int32               // The version of the DS1
-	Width                      int32               // Width of map, in # of tiles
-	Height                     int32               // Height of map, in # of tiles
-	Act                        int32               // Act, from 1 to 5. This tells which act table to use for the Objects list
-	SubstitutionType           int32               // SubstitutionType (layer type): 0 if no layer, else type 1 or type 2
-	NumberOfWalls              int32               // WallNum number of wall & orientation layers used
-	NumberOfFloors             int32               // number of floor layers used
-	NumberOfShadowLayers       int32               // ShadowNum number of shadow layer used
-	NumberOfSubstitutionLayers int32               // SubstitutionNum number of substitution layer used
-	SubstitutionGroupsNum      int32               // SubstitutionGroupsNum number of substitution groups, datas between objects & NPC paths
-	unknown1                   []byte
-	LayerStreamTypes           []d2enum.LayerStreamType
-	unknown2                   uint32
-	NpcIndexes                 []int
+	*ds1Layers
+	Files              []string            // FilePtr table of file string pointers
+	Objects            []Object            // Objects
+	substitutionGroups []SubstitutionGroup // Substitution groups for the DS1
+
+	version          ds1version
+	Act              int32 // Act, from 1 to 5. This tells which Act table to use for the Objects list
+	substitutionType int32 // SubstitutionType (layer type): 0 if no layer, else type 1 or type 2
+	unknown1         []byte
```

Comment:
also, could you do something with this?
I mean, that file created with &DS1{}.Marshal() will be incorrect
so maybe replace with `[unknown1BytesCount]byte` or something like that

Reply:
hmm. thats 8 bytes, so int64.